File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/mediator/lab/Producer.java
// Producer.java

public class Producer
{
	public Producer(String aName)
	{
		myName = aName;
	}

	public int getBid()
	{
		int result = (int) (Math.random() * 100) + 1;

		System.out.println(myName + " bids " + result + ".");

		return result;
	}

	public void produce()
	{
		System.out.println(myName + " produces.");
	}

	private String myName;
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/mediator/lab/producer.cpp
// producer.cpp

#include <iostream>
#include <cstdlib>
#include "producer.h"

producer::producer(const std::string & a_name)
{
	my_name = a_name;
}

int producer::get_bid() const
{
	int result = rand() % 100 + 1;

	std::cout << my_name << " bids " << result << "." << std::endl;

	return result;
}

void producer::produce()
{
	std::cout << my_name << " produces." << std::endl;
}
